src/riak_core_app.erl

```diff
@@ -36,82 +36,112 @@ start(_StartType, _StartArgs) ->
+              "Ring state directory ~p does not exist, " "and could not be created: ~p",
               [RingStateDir, lager:posix_error(RingReason)]),
             throw({error, invalid_ring_state_dir})
-    end,
+    end.
 
+safe_register_cluster_info() ->
     %% Register our cluster_info app callback modules, with catch if
     %% the app is missing or packaging is broken.
     catch cluster_info:register_app(riak_core_cinfo_core),
+    ok.
 
+add_bucket_defaults() ->
     %% add these defaults now to supplement the set that may have been
     %% configured in app.config
     riak_core_bucket:append_bucket_defaults(riak_core_bucket_type:defaults(default_type)),
+    ok.
 
+start_riak_core_sup() ->
     %% Spin up the supervisor; prune ring files as necessary
     case riak_core_sup:start_link() of
         {ok, Pid} ->
-            riak_core:register(riak_core, [{stat_mod, riak_core_stat},
-                                           {permissions, [get_bucket,
-                                                          set_bucket,
-                                                          get_bucket_type,
-                                                          set_bucket_type]}]),
-            ok = riak_core_ring_events:add_guarded_handler(riak_core_ring_handler, []),
-
-            riak_core_capability:register({riak_core, vnode_routing},
-                                          [proxy, legacy],
-                                          legacy,
-                                          {riak_core,
-                                           legacy_vnode_routing,
-                                           [{true, legacy}, {false, proxy}]}),
-            riak_core_capability:register({riak_core, staged_joins},
-                                          [true, false],
-                                          false),
-            riak_core_capability:register({riak_core, resizable_ring},
-                                          [true, false],
-                                          false),
-            riak_core_capability:register({riak_core, fold_req_version},
-                                          [v2, v1],
-                                          v1),
-            riak_core_capability:register({riak_core, security},
-                                          [true, false],
-                                          false),
-            riak_core_capability:register({riak_core, bucket_types},
-                                          [true, false],
-                                          false),
-            riak_core_capability:register({riak_core, net_ticktime},
-                                          [true, false],
-                                          false),
-
-            riak_core_cli_registry:load_schema(),
-            riak_core_cli_registry:register_node_finder(),
-            riak_core_cli_registry:register_cli(),
-
-            riak_core_throttle:init(),
+            ok = register_applications(),
```

Comment:
Ignore my comment about the case statement, I didn't see the other case here due to the formatting of the diff. Nice improvement @jvoegele.